### ISSUE TITLE
alsa broken for speaker-test, aplay and mpv, but not for vlc and jack.  

### ISSUE THREAD

**Issue by magnetophon** [NixOS/nixpkgs#55025]
## Issue description

When playing audio on alsa, I get the following errors:

* speaker-test: ``Setting of hwparams failed: Invalid argument``
* aplay file.wav: ``aplay: set_params:1403: Unable to install hw params:``
* mpv file.wav: ``[ao/alsa] Unable to set hw-parameters: Invalid argument``

Playing audio trough vlc or jack, both using alsa, works fine. 
Oddly, ``speaker-test -c 2`` gives sound as well.

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.5, NixOS, 19.03pre167858.f2a1a4e93be (Koi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(bart): `""`
 - channels(root): `"nixos-19.03pre167858.f2a1a4e93be"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by magnetophon** [comment]
Latest unstable has the same issue:

 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.6, NixOS, 19.03pre168320.2d6f84c1090 (Koi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(bart): `""`
 - channels(root): `"nixos-19.03pre168320.2d6f84c1090"`

**Comment by magnetophon** [comment]
Fixed after upgrading to 19.03pre169108.36f31600749
